third_party/terraform/utils/transport.go

```diff
@@ -177,7 +178,28 @@ func buildReplacementFunc(re *regexp.Regexp, d TerraformResourceData, config *Co
 		}
 	}
 
-	f := func(s string) string {
+	f := func(s string) (sub string) {
+
+		defer func() {
+			// Check if the value returned (the substitute) also contains a replacable string. This must
```

Comment:
s/replacable/replaceable, s/baseUrl's/baseUrls 

I'm not a huge fan of doing this in the replacement function, but I guess it a) does the re search over a smaller value and b) allows you to do the comparison to break chains because you can access the substitution value, so I guess I'm saying I'm conflicted? Maybe more comments or arguments about doing it herer vs recursing in replaceVars or something else, not sure

Reply:
My primary reason for putting it in the builder was so that individual substitutions could be evaluated for self references to avoid the most probable infinite recursing. I had also leaned towards pulling it up to replaceVars but it would mean likely just using a naive depth checking to ensure there wasn't infinite recursion. If you're 👍 with that strategy I'll update.